### PR TITLE
Significant updates for Test 5

### PR DESCRIPTION
**Pull request by kengdoj**
[Preview Changes](https://federalist-78ccea55-7a06-447c-aabf-bd620b207b74.app.cloud.gov/preview/atbcb/icttestingbaseline/kengdoj-changing5/05Changing/)

#350 and #240

Changes to limit to User Controls
- renamed test
- removed previous tests for changes to content
- Name changes
- state and value for user initiated changes
- still no test for role

### CODE REVIEW

File: _baselines/05Changing.md [hunk 1]
```diff
@@ -1,35 +1,107 @@
...
+-   Name Change
+-   Role (test not yet available) 
+-   State 
+-   Value
 
 ### Limitations, Assumptions, or Exceptions
 
--   The state of form elements (checked, unchecked, etc.) are included in this Baseline.
--   User initiated changes and automatic updates are included in this test.
-- If an ARIA live region is used to provide programmatic notification of changes in content, a value of `"off"` for an `aria-live` attribute is ***NOT*** sufficient to provide programmatic notification (whether implicitly or explicitly defined).
```
Comment (cliffbert) [inline _baselines/05Changing.md:21]: Should this test be retained and moved to some other baseline?

File: _baselines/05Changing.md [hunk 2]
```diff
@@ -1,35 +1,107 @@
...
+-   [User interface component (as defined by WCAG)](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components) is a part of the content that is perceived by users as a single control for a distinct function. User interface components include form elements and links as well as components generated by scripts. 
+-   Examples of controls include changes to buttons, form labels, menu items, links.
```
Comment (cliffbert) [inline _baselines/05Changing.md:22]: , and links.

File: _baselines/05Changing.md [hunk 3]
```diff
@@ -1,35 +1,107 @@
...
+-   User-initiated and automatic updates to the name and role of controls are covered in this test. User-initiated changes of state and value of controls are covered in this test. Changes to non-user controls (text, images, etc.) are not covered in this test.
```
Comment (cliffbert) [inline _baselines/05Changing.md:23]: This test covers ... (bullet list)
This test does not cover ...

File: _baselines/05Changing.md [hunk 4]
```diff
@@ -1,35 +1,107 @@
...
    <ul>
-      <li>It may be necessary to use the mouse to determine whether state changes occur on hover or on click.</li>
-      <li>Depending on the component, a change of state may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+      <li>Depending on the control, a change of name may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+	  <li>Examples include entering a response in a form field for country changes the next form field's label from "state" to "province", selecting a control toggles its functionality from sorting ascending to descending, a link appends "Updated" to its name when the linked page is edited.</li>
    </ul>
 
+#### Test Instructions
+<ol id="1TI">
+	<li id="1TI-1">Modify the name of the user control.</li>
```
Comment (cliffbert) [inline _baselines/05Changing.md:38]: Perform an action that modifies the name ... (or similar)

File: _baselines/05Changing.md [hunk 5]
```diff
@@ -1,35 +1,107 @@
...
+#### Test Results
+
+<p id="1TR">If any of the above checks fail, then Baseline Test 5.1-ChangedControlName fails.</p>
+
+### 5.2 Test Procedure for Control State
+
+**Baseline Test ID:** 5.2-ControlState
+#### Identify Content
+<p id="2IC">Identify user-initiated changes to the state of user controls. Examples include changes to buttons, form labels, menu items, links.</p>
+   <ul>
```
Comment (cliffbert) [inline _baselines/05Changing.md:60]: , etc.

File: _baselines/05Changing.md [hunk 6]
```diff
@@ -1,35 +1,107 @@
...
+      <li>Depending on the control, a change of name may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+	  <li>Examples include a disabled "Submit" button is enabled when all required form fields are filled in, a link becomes visible after a user-initiated calculation completes, a check box changes from checked to unchecked, links that are hovered or visited.</li>
```
Comment (cliffbert) [inline _baselines/05Changing.md:62]: change of state?

Comment (cliffbert) [inline _baselines/05Changing.md:62]: bullets?

File: _baselines/05Changing.md [hunk 7]
```diff
@@ -1,35 +1,107 @@
...
+   </ul>
+
+#### Test Instructions
+<ol id="2TI">
+	<li id="2TI-1">Check that the state of the user control is correct in its default state. [SC 4.1.2]</li>
```
Comment (cliffbert) [inline _baselines/05Changing.md:67]: correct when in its default state

File: _baselines/05Changing.md [hunk 8]
```diff
@@ -1,35 +1,107 @@
...
+	<li id="2TI-2">Attributes such as <code>hidden</code>, <code>disabled</code>, and the use of <a href="https://www.w3.org/WAI/standards-guidelines/aria/">ARIA</a> to control component states must be used correctly.</li>
+	<li id="2TI-3">Modify the state of the user control.</li>
```
Comment (cliffbert) [inline _baselines/05Changing.md:69]: perform an action/trigger ...

File: _baselines/05Changing.md [hunk 9]
```diff
@@ -1,35 +1,107 @@
...
+<p id="2TR">If any of the above checks fail, then Baseline Test 5.2-ControlState fails.</p>
+
+### 5.3 Test Procedure for Control Value
+
+**Baseline Test ID:** 5.3-ControlValue
+#### Identify Content
+<p id="3IC">Identify controls that have a value that can be changed by a user. Examples include forms that accept user input to modify another component.</p>
+   <ul>
+      <li>Depending on the control, a change of value may be performed by entering a number, selecting from a list of options, etc.</li>
+	  <li>Examples include adjusting a slider modifies a date range that is highlighted in a calendar, entering numbers in text fields determines a graph's data points, selecting colors determines the appearance of links.</li>
```
Comment (cliffbert) [inline _baselines/05Changing.md:84]: , etc.

File: _baselines/05Changing.md [hunk 10]
```diff
@@ -1,35 +1,107 @@
...
+   </ul>
+
+#### Test Instructions
+<ol id="3TI">
+	<li id="3TI-1">Check that the value of the user control is correct at its default value. [SC 4.1.2]</li>
+	<li id="3TI-2">Modify the value of the user control.</li>
+	<li id="3TI-3">Check that the state of the user control is correct after a change of state. [SC 4.1.2]</li>
```
Comment (cliffbert) [inline _baselines/05Changing.md:91]: state or value?

File: _baselines/05Changing.md [hunk 11]
```diff
@@ -1,35 +1,107 @@
...
+</ol>
+
+#### Test Results
+
+<p id="3TR">If any of the above checks fail, then Baseline Test 5.3-ControlValue fails.</p>
+<!--
+#### Identify Content
+<p id="1IC">Identify changes in presented content (both user driven and automatic). Examples include changes to images, navigation trees, data table sort controls, automatic information updates, form elements, revealed content, etc.</p>
```
Comment (cliffbert) [inline _baselines/05Changing.md:99]: user-driven

File: _baselines/05Changing.md (hunk 2 above)
Comment (kengdoj) [inline _baselines/05Changing.md:22]:
```suggestion
-   Examples of controls include changes to buttons, form labels, menu items, and links.
```

File: _baselines/05Changing.md [hunk 12]
```diff
@@ -1,35 +1,107 @@
...
    <ul>
-      <li>It may be necessary to use the mouse to determine whether state changes occur on hover or on click.</li>
-      <li>Depending on the component, a change of state may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+      <li>Depending on the control, a change of name may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+	  <li>Examples include entering a response in a form field for country changes the next form field's label from "state" to "province", selecting a control toggles its functionality from sorting ascending to descending, a link appends "Updated" to its name when the linked page is edited.</li>
    </ul>
 
+#### Test Instructions
+<ol id="1TI">
+	<li id="1TI-1">Modify the name of the user control.</li>
```
Comment (kengdoj) [inline _baselines/05Changing.md:38]:
```suggestion
	<li id="1TI-1">Perform an action to modify the name of the user control.</li>
```

File: _baselines/05Changing.md [hunk 13]
```diff
@@ -1,35 +1,107 @@
...
+
+#### Test Results
+
+<p id="1TR">If any of the above checks fail, then Baseline Test 5.1-ChangedControlName fails.</p>
+
+### 5.2 Test Procedure for Control State
+
+**Baseline Test ID:** 5.2-ControlState
+#### Identify Content
+<p id="2IC">Identify user-initiated changes to the state of user controls. Examples include changes to buttons, form labels, menu items, links.</p>
```
Comment (kengdoj) [inline _baselines/05Changing.md:59]:
```suggestion
<p id="2IC">Identify user-initiated changes to the state of user controls. Examples include changes to buttons, form labels, menu items, and links.</p>
```

File: _baselines/05Changing.md [hunk 14]
```diff
@@ -1,35 +1,107 @@
...
+#### Identify Content
+<p id="2IC">Identify user-initiated changes to the state of user controls. Examples include changes to buttons, form labels, menu items, and links.</p>
+   <ul>
+      <li>Depending on the control, a change of name may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+	  <li>Examples include a disabled "Submit" button is enabled when all required form fields are filled in, a link becomes visible after a user-initiated calculation completes, a check box changes from checked to unchecked, links that are hovered or visited.</li>
+   </ul>
+
+#### Test Instructions
+<ol id="2TI">
+	<li id="2TI-1">Check that the state of the user control is correct in its default state. [SC 4.1.2]</li>
```
Comment (kengdoj) [inline _baselines/05Changing.md:67]:
```suggestion
	<li id="2TI-1">Check that the state of the user control is correct when in its default state. [SC 4.1.2]</li>
```

File: _baselines/05Changing.md [hunk 15]
```diff
@@ -1,35 +1,107 @@
...
+#### Identify Content
+<p id="3IC">Identify controls that have a value that can be changed by a user. Examples include forms that accept user input to modify another component.</p>
+   <ul>
+      <li>Depending on the control, a change of value may be performed by entering a number, selecting from a list of options, etc.</li>
+	  <li>Examples include adjusting a slider modifies a date range that is highlighted in a calendar, entering numbers in text fields determines a graph's data points, and selecting colors determines the appearance of links.</li>
+   </ul>
+
+#### Test Instructions
+<ol id="3TI">
+	<li id="3TI-1">Check that the value of the user control is correct at its default value. [SC 4.1.2]</li>
```
Comment (kengdoj) [inline _baselines/05Changing.md:89]:
```suggestion
	<li id="3TI-1">Check that the value of the user control is correct when at its default value. [SC 4.1.2]</li>
```

File: _baselines/05Changing.md [hunk 16]
```diff
@@ -1,35 +1,107 @@
...
+
+<p id="1TR">If any of the above checks fail, then Baseline Test 5.1-ChangedControlName fails.</p>
+
+### 5.2 Test Procedure for Control State
+
+**Baseline Test ID:** 5.2-ControlState
+#### Identify Content
+<p id="2IC">Identify user-initiated changes to the state of user controls. Examples include changes to buttons, form labels, menu items, and links.</p>
+   <ul>
+      <li>Depending on the control, a change of name may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
```
Comment (kengdoj) [inline _baselines/05Changing.md:61]:
```suggestion
      <li>Depending on the control, a change of state may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
```

File: _baselines/05Changing.md (hunk 6 above)
Comment (kengdoj) [reply _baselines/05Changing.md:62]: thanks, changed name to state on line 61.

**Review by cliffbert — APPROVED** [review]

File: _baselines/05Changing.md [hunk 17]
```diff
@@ -1,35 +1,107 @@
...
+-   Value
 
 ### Limitations, Assumptions, or Exceptions
 
--   The state of form elements (checked, unchecked, etc.) are included in this Baseline.
--   User initiated changes and automatic updates are included in this test.
-- If an ARIA live region is used to provide programmatic notification of changes in content, a value of `"off"` for an `aria-live` attribute is ***NOT*** sufficient to provide programmatic notification (whether implicitly or explicitly defined).
+-   [User interface component (as defined by WCAG)](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components) is a part of the content that is perceived by users as a single control for a distinct function. User interface components include form elements and links as well as components generated by scripts. 
+-   Examples of controls include changes to buttons, form labels, menu items, and links.
+-   User-initiated and automatic updates to the name and role of controls are covered in this test. User-initiated changes of state and value of controls are covered in this test. Changes to non-user controls (text, images, etc.) are not covered in this test.
```
Comment (kengdoj) [inline _baselines/05Changing.md:23]:
```suggestion
-   This test covers User-initiated and automatic updates to the name and role of controls are covered in this test. User-initiated changes of state and value of controls are covered in this test. Changes to non-user controls (text, images, etc.) are not covered in this test.
```

File: _baselines/05Changing.md [hunk 18]
```diff
@@ -1,35 +1,107 @@
 ---
-title: "5. Changing Content"
+title: "5. User Controls"
```
Comment (drewnielson) [inline _baselines/05Changing.md:2]:
```suggestion
title: "5. User Control Changes"
```

File: _baselines/05Changing.md [hunk 19]
```diff
@@ -1,35 +1,107 @@
...
 
 ### Test Method Rationale
 
-The purpose of this Baseline test is to identify visual changes to content and component states that occur without a page refresh and to correlate those changes with equivalent programmatic and/or textual information. Notification of such changes must be provided.
--   Content changes could include component state changes (e.g., closed/open sections, sort order, collapsed/expanded menus), dynamic and/or calculated values (e.g., "34 characters remaining"), and automatically updating content (e.g., stock prices).
--   Content that changes in the course of a user's interaction with the content (including, but not limited to changes in text, component states, structure, relationships, or functionality) must provide a means by which the change in content is programmatically determinable or is otherwise available in text.
+The purpose of this Baseline test is to check the following accessibility properties of user controls:
+-   Name Change
+-   Role (test not yet available)
```
[the remark this answers is not in the document]
Comment (kengdoj) [reply _baselines/05Changing.md:15]: Role tests have not been added, yet.

File: _baselines/05Changing.md [hunk 20]
```diff
@@ -1,35 +1,107 @@
...
+<p id="1IC">Identify changes (both user driven and automatic) in the functionality, appearance or label of user controls. </p>
    <ul>
-      <li>It may be necessary to use the mouse to determine whether state changes occur on hover or on click.</li>
-      <li>Depending on the component, a change of state may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+      <li>Depending on the control, a change of name may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+	  <li>Examples include entering a response in a form field for country changes the next form field's label from "state" to "province", selecting a control toggles its functionality from sorting ascending to descending, a link appends "Updated" to its name when the linked page is edited.</li>
```
[the remark this answers is not in the document]
Comment (kengdoj) [reply _baselines/05Changing.md:33]: removed the changes to controls part here

**Comment by kengdoj** [comment]
Please take another look.

File: _baselines/05Changing.md [hunk 21]
```diff
@@ -46,28 +41,36 @@ The purpose of this Baseline test is to check the following accessibility proper
...
-	<li id="2TI-3">Perform an action to modify the state of the user control.</li>
-	<li id="2TI-4">Check that the state of the user control is correct after a change of state. [SC 4.1.2]</li>
+	<li id="2TI-1">Check that the state of the user control is correct. Attributes such as <code>hidden</code>, <code>disabled</code>, and the use of <a href="https://www.w3.org/WAI/standards-guidelines/aria/">ARIA</a> to control component states must be used correctly.[SC 4.1.2]</li>
+	<li id="2TI-2">If the state of the user control changes with use of the application, check that the state of the user control is correct after a change of state. [SC 4.1.2]
+		<ul>
+			<li>Depending on the control, a change of state may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+			<li>Examples include a disabled "Submit" button is enabled when all required form fields are filled in, a link becomes visible after a user-initiated calculation completes, a check box changes from checked to unchecked, links that are hovered or visited.</li>
```
Comment (cliffbert) [inline _baselines/05Changing.md:71]: "links that are hovered over"?

**Review by cliffbert — APPROVED** [review]

File: _baselines/05Changing.md (hunk 21 above)
Comment (bruce-usab) [reply _baselines/05Changing.md:71]: Good catch!   The browser can tell between (recently) visited and unvisited links, but does that count as a state change?

**Review by bruce-usab — APPROVED** [review]
Editorial suggestion, current:

> Examples include a disabled "Submit" button is enabled when all required form fields are filled in, a link becomes visible after a user-initiated calculation completes, a check box changes from checked to unchecked, links that are hovered or visited.

Proposed:
> Examples include: a disabled "Submit" button is enabled when all required form fields are filled in; content which becomes visible after a user-initiated calculation completes; a check box changes from checked to unchecked; and visited links being distinguished from unvisited links.

File: _baselines/05Changing.md [hunk 22]
```diff
@@ -1,72 +1,104 @@
...
 
 ### Test Method Rationale
 
-The purpose of this Baseline test is to identify visual changes to content and component states that occur without a page refresh and to correlate those changes with equivalent programmatic and/or textual information. Notification of such changes must be provided.
--   Content changes could include component state changes (e.g., closed/open sections, sort order, collapsed/expanded menus), dynamic and/or calculated values (e.g., "34 characters remaining"), and automatically updating content (e.g., stock prices).
--   Content that changes in the course of a user's interaction with the content (including, but not limited to changes in text, component states, structure, relationships, or functionality) must provide a means by which the change in content is programmatically determinable or is otherwise available in text.
+The purpose of this Baseline test is to check the following accessibility properties of user controls:
+-   Name
+-   Role (test not yet available)
```
[the remark this answers is not in the document]
Comment (drewnielson) [reply _baselines/05Changing.md:15]: I think we can leave out reference to whether a test is available (that we know of). We didn't make the same disclaimer for Parsing or Flashing, so probably not necessary here either.

File: _baselines/05Changing.md [hunk 23]
```diff
@@ -1,72 +1,104 @@
...
+**Baseline Test ID:** 5.1-ChangedControlName
 
-**Baseline Test ID:** 5.1-ChangeContent
 #### Identify Content
-<p id="1IC">Identify changes in presented content (both user driven and automatic). Examples include changes to images, navigation trees, data table sort controls, automatic information updates, form elements, revealed content, etc.</p>
-   <ul>
-      <li>It may be necessary to use the mouse to determine whether state changes occur on hover or on click.</li>
-      <li>Depending on the component, a change of state may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
-   </ul>
+<p id="1IC">Identify user controls for a distinct function. Exclude forms and links as these are covered by [Baseline 10. Forms](../10Forms), [Baseline 14. Links](../14Links), respectively.</p>
```
Comment (cliffbert) [inline _baselines/05Changing.md:29]: "... for distinct functions"? unless we mean test one function at a time, which may indeed be the case.

File: _baselines/05Changing.md [hunk 24]
```diff
@@ -1,72 +1,104 @@
...
+<p id="3IC">Identify user controls for a distinct function. Examples include changes to forms, links, and toggle controls.</p>
+  
+
+#### Test Instructions
+<ol id="2TI">
+	<li id="2TI-1">Check that the state of the user control is correct. Attributes such as <code>hidden</code>, <code>disabled</code>, and the use of <a href="https://www.w3.org/WAI/standards-guidelines/aria/">ARIA</a> to control component states must be used correctly.[SC 4.1.2]</li>
+	<li id="2TI-2">If the state of the user control changes with use of the application, check that the state of the user control is correct after a change of state. [SC 4.1.2]
+		<ul>
+			<li>Depending on the control, a change of state may be triggered by various actions, such as changing values or states of other components, toggling a function, entering data in the component, mouseover, etc.</li>
+			<li>Examples include a disabled "Submit" button is enabled when all required form fields are filled in, a link becomes visible after a user-initiated calculation completes, a check box changes from checked to unchecked, links that are hovered or visited.</li>
```
Comment (kengdoj) [inline _baselines/05Changing.md:71]:
```suggestion
			<li>Examples include a disabled "Submit" button is enabled when all required form fields are filled in, a link becomes visible after a user-initiated calculation completes, and a check box changes from checked to unchecked.</li>
```

**Review by bruce-usab — APPROVED** [review]
Looks good!

File: _baselines/05Changing.md (hunk 22 above)
[the remark this answers is not in the document]
Comment (kengdoj) [reply _baselines/05Changing.md:15]: Write the test to check that role is a valid, is it correct for its purpose?

**Review by bruce-usab — APPROVED** [review]
There is a current thread/issue in WCAG that about links that (visually) look like a button but do not have ARIA role or other means to expose that information to screen reading software.  I think that discussion relates to this Test 5.  (But also that Test 5 is fine.) 

As I <q>read the room</q> the consensus seems to be that it's okay for a link to present as button (but maybe not vice-versa).

If you are interested:  https://github.com/w3c/wcag/issues/2142

**Pull request merged**